src/Components/test/testassets/Ignitor/BlazorClient.cs

```diff
@@ -75,55 +163,92 @@ public async Task<bool> ConnectAsync(Uri uri, bool prerendered)
             if (prerendered)
             {
                 CircuitId = await GetPrerenderedCircuitIdAsync(uri);
-                return await HubConnection.InvokeAsync<bool>("ConnectCircuit", CircuitId);
+                var result = false;
+                await ExpectRenderBatch(async () => result = await HubConnection.InvokeAsync<bool>("ConnectCircuit", CircuitId));
+                return result;
             }
             else
             {
-                CircuitId = await HubConnection.InvokeAsync<string>("StartCircuit", uri, uri);
+                await ExpectRenderBatch(
+                    async () => CircuitId = await HubConnection.InvokeAsync<string>("StartCircuit", new Uri(uri.GetLeftPart(UriPartial.Authority)), uri),
+                    TimeSpan.FromSeconds(10));
                 return CircuitId != null;
             }
+        }
 
-            void OnBeginInvokeJS(int asyncHandle, string identifier, string argsJson)
+        private void OnBeginInvokeJS(int asyncHandle, string identifier, string argsJson)
+        {
+            try
             {
                 JSInterop?.Invoke(asyncHandle, identifier, argsJson);
+
+                NextJSInteropReceived?.Completion?.TrySetResult(null);
+            }
+            catch (Exception e)
+            {
+                NextJSInteropReceived?.Completion?.TrySetResult(e);
```

Comment:
Is this right? Should this be `TrySetException`? Is the idea that whomstve'ever is looking at this will do a cast to see if it is a success or failure?

Reply:
No, I made a mistake here.